LeaderboardBackend/Models/Entities/ApplicationContext.cs

```diff
@@ -1,14 +1,25 @@
+using System.Collections.Concurrent;
 using System.Reflection;
 using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Options;
 using Npgsql;
 
 namespace LeaderboardBackend.Models.Entities;
 
 public class ApplicationContext : DbContext
 {
     public const string CASE_INSENSITIVE_COLLATION = "case_insensitive";
-    public ApplicationContext(DbContextOptions<ApplicationContext> options)
-        : base(options) { }
+
+    // the HashCode is calculated with all of the config's property because PostgresConfig is a record
+    private static readonly ConcurrentDictionary<PostgresConfig, NpgsqlDataSource> _dataSourceCache = new();
```

Comment:
I don't think this is necessary. We only need one datasource for the whole application.